pennylane/qchem/observable_hf.py

```diff
@@ -145,6 +143,11 @@ def qubit_observable(o_ferm, cutoff=1.0e-12, mapping="jordan_wigner"):
         qubits = len(o_ferm.wires)
         h = qml.bravyi_kitaev(o_ferm, qubits, ps=True, tol=cutoff)
 
+    if list(h.wires) != sorted(list(h.wires)):
+        h = PauliSentence(
+            sorted(h.items(), key=lambda item: max(item[0].wires.tolist(), default=0))
```

Comment:
What does this default do here? How is this different from the previous empty PauliWord thing you used?

Reply:
1. The previous method of creating a term applied on all wires, with 0 coefficient, does not work in some edge cases because `.simplify()` makes sure that its effect is removed. 

2. The default only ensure that you don't get an error with `sorted` when you have an identity term in the Hamiltonian, with no wires. But, the important upgrade is using `max`, which sorts the terms such that each new term has at least 1 wire that is larger than the maximum wire index of the previous term. For example, that ensures the following sorting: 
`[[], [0, 1, 2, 3], [0, 1, 2, 4], [0, 1, 2, 3, 5]]` while the original sort was `[[], [0, 1, 2, 3], [0, 1, 2, 3, 5], [0, 1, 2, 4]]`.